CMakeLists.txt

```diff
@@ -159,74 +161,60 @@ ELSE()
-
-IF ("${RTTR_PREFIX}" STREQUAL "")
-    IF ("${CMAKE_INSTALL_PREFIX}" STREQUAL "")
-        SET(RTTR_PREFIX .)
-    ELSE()
-        SET(RTTR_PREFIX ${CMAKE_INSTALL_PREFIX})
-    ENDIF()
-ENDIF()
-
-SET(RTTR_PREFIX ${RTTR_PREFIX} CACHE STRING "Path prefix")
-SET(RTTR_BINDIR ${RTTR_BINDIR} CACHE STRING "Directory for the binaries")
-SET(RTTR_DATADIR ${RTTR_DATADIR} CACHE STRING "Data directory")
-SET(RTTR_GAMEDIR ${RTTR_GAMEDIR} CACHE STRING "Game data directory")
-SET(RTTR_LIBDIR ${RTTR_LIBDIR} CACHE STRING "Directory for shared binaries")
-SET(RTTR_DRIVERDIR ${RTTR_DRIVERDIR} CACHE STRING "Directory for driver objects")
-SET(RTTR_SRCDIR ${CMAKE_CURRENT_SOURCE_DIR} CACHE INTERNAL "")
-
-IF (NOT "${RTTR_PREFIX}" STREQUAL "" AND "${RTTR_BINDIR}" STREQUAL "")
-	SET(RTTR_BINDIR ${RTTR_PREFIX}/bin CACHE STRING "Directory for the binaries" FORCE)
-	MESSAGE(STATUS "Set RTTR_BINDIR to ${RTTR_BINDIR}")
-ENDIF ()
-
-IF (NOT "${RTTR_PREFIX}" STREQUAL "" AND "${RTTR_DATADIR}" STREQUAL "")
-	SET(RTTR_DATADIR ${RTTR_PREFIX}/share/s25rttr CACHE STRING "Data directory" FORCE)
-	MESSAGE(STATUS "Set RTTR_DATADIR to ${RTTR_DATADIR}")
-ENDIF()
-
-IF (NOT "${RTTR_DATADIR}" STREQUAL "" AND "${RTTR_GAMEDIR}" STREQUAL "")
-	SET(RTTR_GAMEDIR ${RTTR_DATADIR} CACHE STRING "Game data directory" FORCE)
-	MESSAGE(STATUS "Set RTTR_GAMEDIR to ${RTTR_DATADIR}")
-ENDIF()
-
-IF (NOT "${RTTR_DATADIR}" STREQUAL "" AND "${RTTR_LIBDIR}" STREQUAL "")
-	SET(RTTR_LIBDIR ${RTTR_DATADIR} CACHE STRING "Directory for shared binaries" FORCE)
-	MESSAGE(STATUS "Set RTTR_LIBDIR to ${RTTR_LIBDIR}")
-ENDIF()
-
-IF (NOT "${RTTR_LIBDIR}" STREQUAL "" AND "${RTTR_DRIVERDIR}" STREQUAL "")
-	SET(RTTR_DRIVERDIR "${RTTR_LIBDIR}/driver" CACHE STRING "Directory for driver objects" FORCE)
-	MESSAGE(STATUS "Set RTTR_DRIVERDIR to ${RTTR_DRIVERDIR}")
+# Set paths
+# All paths are relative to the install-prefix.
+# Example: BINDIR="bin", DATADIR="data" executed in "/foo/bar/bin/rttr.exe" -> FULL_DATADIR=/foo/bar/data
+IF(NOT "${RTTR_PREFIX}" STREQUAL "")
+	MESSAGE(FATAL_ERROR "Using RTTR_PREFIX was removed and the meaning of the paths have changed. Clear RTTR_PREFIX to disable this warning")
 ENDIF()
 
-# Change DESTDIR if required. Note: This is a variable used by cmake and install, do not rename!
-# -> https://cmake.org/cmake/help/v3.0/variable/CMAKE_INSTALL_PREFIX.html
-IF(NOT "$ENV{CMAKE_INSTALL_PREFIX}" STREQUAL "")
-	SET(ENV{DESTDIR} "$ENV{CMAKE_INSTALL_PREFIX}" CACHE STRING "" FORCE)
-	MESSAGE(STATUS "Set DESTDIR to $ENV{DESTDIR}")
+IF ("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
+	# Windows uses one directory only
+	SET(RTTR_BINDIR "." CACHE INTERNAL "")
+	SET(RTTR_DATADIR "." CACHE INTERNAL "")
+	SET(RTTR_GAMEDIR "." CACHE INTERNAL "")
+	SET(RTTR_LIBDIR "." CACHE INTERNAL "")
+	SET(RTTR_DOCDIR "." CACHE INTERNAL "")
+	SET(RTTR_DRIVERDIR "./driver" CACHE INTERNAL "")
+ELSE()
+	include(GNUInstallDirs)
+	SET(RTTR_BINDIR "${CMAKE_INSTALL_BINDIR}" CACHE STRING "Directory for the binaries")
```

Comment:
These should not be cachable. Users are now able to set `CMAKE_INSTALL_BINDIR` and `RTTR_BINDIR` independently, which doesn't really make sense. Only the GNUInstallDirs variables should be exposed, everything else not, imo. So if you use RTTR_DATADIR for the preprocesser, you don't want to cache and reverse-overwrite it, instead you want something like:

```
SET(RTTR_DATADIR "${CMAKE_INSTALL_FULL_DATADIR}/s25rttr/S2")
```
which ensures it's a proper full path.